blog/_src/posts/2017-01-02-fall-2016-pl-junior-retrospective.md

```diff
@@ -0,0 +1,119 @@
+inductive data definitions and eliminations. There were some extremely
+large type signatures. While this exercise boosted our confidence that
+we could read Serious Dependent Types™ papers, it also gave evidence
+that our fears of incomprehensibility were not completely unfounded.
+
+This issue appeared before in our discussion of Girard’s Paradox. In
+the discussion of the paradox, we got stuck when we tried to
+understand the very complex term that inhabited the bottom type.
+Dependent typing, and discussions thereof, allow very rich,
+meaningful, and complex types that are as complex as the code that
+they abstract over. While we are used to understanding these
+structures in code, parsing a complex type and its fundamental meaning
+gave us considerable difficulty.
+
+## Thoughts on the format
+
+Our meetings this semester were all of the form “we’ll watch this
+lecture or read this chapter, and then discuss it next week.” Next
+semester we would like to go back to presenting each week. We feel
+doing presentations forces the presenter to reach a deeper
+understanding of the material. This semester we got a relatively
+shallow understanding of a broad area. A deeper understanding with a
+narrower focus may be more beneficial (or complementary).
+
+[[Sam’s defense as Grand Convener of PL Junior: Once we picked dependent
+types as a topic, doing presentations was not an option. We didn’t
+have the expertise in the area to pick out different sub-topics and
+papers suitable for presentations. And, since we were short-handed (4
+people each week), we would be presenting once a month!]]
+
+If we continue learning more about dependent types it would be by: 1)
+Doing more reading, such as the [ATTAPL] [9] chapter or the programming
+in Martin-Löf’s type theory material. 2) Actually trying to implement
+some of the things we’ve learned this semester 3) Playing around with
+more of the various dependent type systems and theorem provers out
+there
+
+For future pl junior cohorts or anyone else in learning about
+dependent types: Pi-for-all is useful material, but could be condensed
+into two weeks (for example, by watching the lectures at 1.5x speed)
+instead of four. Don’t worry about Type-in-Type. The Epigram material
+is OK but you might be better served looking at ATTAPL first. At some
+point, you will have to read the dense papers, but pi-for-all is a
+good introduction.
+
+[1]: https://mitpress.mit.edu/books/types-and-programming-languages
+[2]: https://github.com/sweirich/pi-forall
+[3]: https://eb.host.cs.st-andrews.ac.uk/writings/thesis.pdf
+[4]: http://purelytheoretical.com/sywtltt.html
+[5]: http://jozefg.bitbucket.org/posts/2015-08-14-learn-tt.html
+[6]: https://en.wikipedia.org/wiki/System_U#Girard.27s_paradox
+[7]: https://www.cs.cmu.edu/~kw/scans/hurkens95tlca.pdf
+[8]: https://www.cs.uoregon.edu/research/summerschool/
+[9]: https://www.cis.upenn.edu/~bcpierce/attapl/
+
+<!-- more -->
```

Comment:
move this to line 12 (otherwise your `make preview` or `raco frog -bp` will be surprising)